docs/rules/object-curly-newline.md

```diff
@@ -0,0 +1,473 @@
+# require or disallow line breaks inside braces (object-curly-newline)
+
+(fixable) The `--fix` option on the [command line](../user-guide/command-line-interface#fix) automatically fixes problems reported by this rule.
+
+A number of style guides require or disallow line breaks inside of object braces and other tokens.
+This rule applies to both object literals and destructuring assignments.
+
+## Rule Details
+
+This rule enforces consistent line breaks inside braces.
+
+## Options
+
+```json
+{
+    "object-curly-newline": ["error", {"multiline": true}]
+}
+```
+
+This rule has a string or an object option:
+
+* `"always"` - requires line breaks always.
+* `"never"` - disallows line breaks always.
+* An object - requires line breaks if any of the following condition is satisfied. Otherwise, disallows line breaks.
+    * `multiline` (default is `false`) - requires line breaks if the content is multiline. If this is `false`, this condition is disabled.
```

Comment:
It's all characters inside of braces -- all properties. In my head, there was also `array-bracket-newline` (all elements and parentheses) when I wrote this.